### PR TITLE
Add a note for `Mojo::Base` in Documentation for Tests

### PR DESCRIPTION
**Pull request by josegomezr**
- Explain that it includes default best practices for perl (warnings, strict, utf8, etc.).

- Add a link to `Mojo::Base` upstream docs.

### CODE REVIEW

File: docs/WritingTests.asciidoc [hunk 1]
```diff
@@ -51,13 +51,15 @@ Here is an example in Perl:
 
 [source,perl]
 -------------------------------------------------------------------
-use Mojo::Base "basetest";
+use Mojo::Base "basetest"; # See note <1>
```
Comment (okurz) [inline docs/WritingTests.asciidoc:54]: Hm, keeping that in the code example means that people either copy the comment verbatim with the dangling reference or they need to remove that comment. How about we just keep the note below in line 62 and you don't change the code example itself?

File: docs/WritingTests.asciidoc [hunk 2]
```diff
@@ -51,13 +51,15 @@ Here is an example in Perl:
...
+
```
Comment (okurz) [inline docs/WritingTests.asciidoc:55]: unrelated change. I would prefer to not add that blank line.

**Review by okurz — CHANGES_REQUESTED** [review]

**Comment by josegomezr** [comment]
Removed the blank line & the `<>` callout in favor of a note after the code sample.

File: docs/WritingTests.asciidoc (hunk 2 above)
Comment (kalikiana) [reply docs/WritingTests.asciidoc]: Can we lint that?

Comment (kalikiana) [reply docs/WritingTests.asciidoc]: Not in this branch, just wondering since it's a very common point that comes up

**Review by kalikiana — APPROVED** [review]
Nice!
